### PR TITLE
style: adds base/reset CSS file

### PR DESCRIPTION
**Pull request by anechol**
# Description

Adds a shared base/reset file to all components. This will allow us to get rid of repetitive style declarations within components such as font styles and typical reset styles, as well as address some minor issues like the `[hidden]` attribute being overridden by more specific selectors. There are no additional network requests made for these added styles. The styles are bundled up and combined into one stylesheet when the component is built.

If there's another approach that might work better, keeping in mind to not interfere with styles outside of the components **and** keeping specificity as low as possible, please share your ideas!

[DSS-863](https://kajabi.atlassian.net/browse/DSS-863)

## Type of change

- [x] Style (adds, removes, or refactors component styling)

# How Has This Been Tested?

Please make sure all components look as expected. The base file has also been updated with _most_ of the needed semantic tokens.

- [ ] unit tests
- [ ] e2e tests
- [ ] accessibility tests
- [x] tested manually
- [ ] other:

**Test Configuration**:

- Pine versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

[DSS-863]: https://kajabi.atlassian.net/browse/DSS-863?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/core/src/components/pds-accordion/stories/pds-accordion.stories.js [hunk 1]
```diff
@@ -13,21 +13,17 @@ export default {
...
+      <pds-icon name="product"></pds-icon>
+      <span style="display: inline-block; margin-left: 8px;">Products</span>
+    </pds-box>
+    <pds-box direction="column" gap="xs" padding="xs">
+      <pds-link variant="plain" href="#">All products</pds-link>
+      <pds-link variant="plain" href="#">Courses</pds-link>
+      <pds-link variant="plain" href="#">Coaching</pds-link>
+      <pds-link variant="plain" href="#">Community</pds-link>
+      <pds-link variant="plain" href="#">Podcasts</pds-link>
+    </pds-box>
```
Comment (ju-Skinner) [inline libs/core/src/components/pds-accordion/stories/pds-accordion.stories.js:26]: Did you intend to commit this with this effort?

File: libs/core/src/global/styles/base.scss [hunk 2]
```diff
@@ -0,0 +1,121 @@
+// TODO: replace core tokens with semantic
```
Comment (ju-Skinner) [inline libs/core/src/global/styles/base.scss:1]: do we still need this todo?

File: libs/core/src/components/pds-accordion/stories/pds-accordion.stories.js (hunk 1 above)
Comment (anechol) [reply libs/core/src/components/pds-accordion/stories/pds-accordion.stories.js:26]: Yes this was on purpose. Just removing an unneeded `pds-row`.

File: libs/core/src/global/styles/base.scss (hunk 2 above)
Comment (anechol) [reply libs/core/src/global/styles/base.scss:1]: Yes, I still have some semantic type tokens to create. I should be getting that done today, but those are separate tickets.

File: libs/core/.stylelintignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
 src/global/styles/_fonts.scss
-src/global/styles/tokens/semantic/*.scss
\ No newline at end of file
+src/global/styles/tokens/semantic/*.scss
+src/global/styles/base.scss
```
Comment (pixelflips) [inline libs/core/.stylelintignore]: nitpick: file missing new line at end

File: libs/core/src/global/styles/app.scss [hunk 4]
```diff
@@ -1,2 +1,2 @@
-@use 'fonts';
 @use 'tokens/index';
+@use 'fonts';
```
Comment (pixelflips) [inline libs/core/src/global/styles/app.scss]: nitpick: file missing new line at end

**Review by pixelflips — APPROVED** [review]
A couple of non-blocking nitpicks, otherwise LGTM! Great work!

File: libs/core/src/global/styles/base.scss [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+/*
+  Josh's Custom CSS Reset
+  https://www.joshwcomeau.com/css/custom-css-reset/
+*/
+:where(*:not(slot), *:not(slot)::before, *:not(slot)::after) {
```
Comment (QuintonJason) [inline libs/core/src/global/styles/base.scss:8]: 🏆

**Review by QuintonJason — APPROVED** [review]

**Review by pixelflips — APPROVED** [review]

**Review by QuintonJason — APPROVED** [review]
Great stuff!

**Pull request merged**